src/olympia/abuse/admin.py

```diff
@@ -185,6 +183,38 @@ def get_actions(self, request):
             actions.pop('mark_as_suspicious')
         return actions
 
+    def get_search_fields(self, request):
+        """
+        Return search fields according to the type filter.
+        """
+        type_ = request.GET.get('type')
+        if type_ == 'addon':
+            search_fields = (
+                'addon__name__localized_string', 'addon__slug', 'addon_name',
+                '=guid', 'message', 'addon__id',
```

Comment:
is `addon_id` more efficient than `addon_id` (and should it be `'=addon_id'` or `'addon_id__exact'`?)

Reply:
Yes, but that doesn't work :( Django doesn't let me, and I didn't want to add even more custom code to `get_search_fields()` (I originally had, but then with the other recent changes that was a little too much in this method).

I haven't looked at the consequences of using `addon__id` besides the fact it's working but hopefully the resulting query is not that bad.

Reply:
what doesn't work?  `=` is old syntax, supported here: https://github.com/mozilla/addons-server/pull/11438/files#diff-673ddf9dd55a8432ead7781c781b0d6fR39 and `__exact` is the new syntax, supported in the same method.

Reply:
I'm talking about using `addon_id`. It raises an exception, because it's trying to use the field as a string instead of an id. Weirdly, it works with `addon__id` and `id` - django docs warn that this is meant for text search originally... 

I don't think I need `=`/`exact` with it but I'll check. I did try and it didn't solve the original issue (brokeness with `addon_id`)

Reply:
- With `=addon__id`, it seems to be working
- With `addon__id__exact`, `TestAbuse.test_search_addon` fails with `ValueError: invalid literal for int() with base 10: 'sterious'`
- With `=addon_id`, that test fails with `django.core.exceptions.FieldError: Related Field got invalid lookup: iexact`
- With `addon_id__exact`, that test fails with `ValueError: invalid literal for int() with base 10: 'sterious'`

Similar results with `user_id` (using `TestAbuse.test_search_user`).

Conclusion, I'm going to keep traversing the relation, but use the `=` prefix.

Reply:
Note that as soon as we have a list of search terms (more than 1 term), all numeric, we switch to the `get_search_id_field()` where we don't have that problem.